test/unit/spec/function_output/kbaseReportView-spec.js

```diff
@@ -1,8 +1,311 @@
+            description: 'Simple test file 1',
+            handle: 'KBH_658368',
+            label: '',
+            name: 'file_number_1.txt'
+        }],
+        html_links: [{
+            URL: 'https://ci.kbase.us/services/shock-api/node/bf67e724-2025-477a-9cc9-868e71e89cbf',
+            description: 'Report page 1',
+            handle: 'KBH_658369',
+            label: '',
+            name: 'page1.html'
+        }, {
+            URL: 'https://ci.kbase.us/services/shock-api/node/5caa6be0-0e71-4fab-92a0-8ede347b4a5b',
+            description: 'Report page 2',
+            handle: 'KBH_658370',
+            label: '',
+            name: 'page2.html'
+        }],
+        html_window_height: null,
+        objects_created: [{
+            ref: '4/5/6',
+            description: 'an object'
+        }],
+        summary_window_height: null,
+        text_message: 'Here is an example report',
+        warnings: []
+    };
+
+    const CREATED_OBJECTS_INFO = [
+        [5, 'SomeObject', 'KBaseGenomes.Genome-1.0', new Date().getTime(), 6, 'some_user', 4, 'some_workspace'],
+    ];
+
+    function mockReportLookup(reportData) {
+        Mocks.mockJsonRpc1Call({
+            url: Config.url('workspace'),
+            body: /get_objects2/,
+            response: {
+                data: [{
+                    data: reportData
+                }]
+            }
+        });
+    }
+
+    function mockCreatedObjectInfo(objectInfo) {
+        Mocks.mockJsonRpc1Call({
+            url: Config.url('workspace'),
+            body: /get_object_info_new/,
+            response: objectInfo
+        });
+    }
+
+    /**
+     * @param {object} panels should have these properties with boolean values
+     * - createdObjects
+     * - htmlPanel
+     * - summary
+     * - htmlLinks
+     * - fileLinks
+     * @param {object} $node - the jquery node hosting the report viewer
+     */
+    function verifyPanelPresence(panels, $node) {
+        const panelSelectors = {
+            createdObjects: '.kb-created-objects',
+            htmlPanel: '[data-element="html-panel"]',
+            summary: '[data-element="summary-section"]',
+            htmlLinks: '[data-element="downloadable-html"]',
+            fileLinks: '[data-element="downloadable-files"]'
+        };
+        for (const [panel, selector] of Object.entries(panelSelectors)) {
+            let expectation = 0;
+            if (panels[panel]) {
+                expectation = 1;
+            }
+            expect($node.find(selector).length).toEqual(expectation);
+        };
+    }
+
     describe('The kbaseReportView widget', () => {
+        beforeAll(() => {
+            Jupyter.narrative = {
+                getAuthToken: () => 'fakeToken'
+            };
+        });
+
+        afterAll(() => {
+            Jupyter.narrative = null;
+        });
+
+        beforeEach(function () {
+            jasmine.Ajax.install();
+            this.$node = $('<div>');
+            Mocks.mockJsonRpc1Call({
```

Comment:
This is the same each time - it's only used to build the links to the stored HTML reports in an iframe.